src/System.Runtime/tests/System/HashCodeTests.cs

```diff
@@ -0,0 +1,91 @@
+    [InlineData(0x4d79177dU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU)]
+    [InlineData(0x59d79205U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U)]
+    [InlineData(0x49585aaeU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U)]
+    [InlineData(0x2f005ff1U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U)]
+    [InlineData(0x0ce339bdU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U)]
+    [InlineData(0xb31bd2ffU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U, 0x33323130U)]
+    [InlineData(0xa821efa3U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U, 0x33323130U, 0x62617a79U)]
+    public static void HashCode_Add(uint expected, params uint[] vector)
+    {
+        var hc = new HashCode();
+        for (var i = 0; i < vector.Length; i++)
+            hc.Add(vector[i]);
+
+        Assert.Equal(expected, unchecked((uint)hc.ToHashCode()));
+    }
+
+    [Fact]
+    public static void HashCode_Add_Generic()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(100.2);
+
+        Assert.Equal(-273013950, hc.ToHashCode());
+    }
+
+    [Fact]
+    public static void HashCode_Add_GenericEqualityComparer()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add("Hello", new ConstComparer());
+
+        Assert.NotEqual(-1844029331, hc.ToHashCode());
+    }
+
+    [Fact]
+    public static void HashCode_Combine()
+    {
+        var hcs = new[]
+        {
+            HashCode.Combine(1),
+            HashCode.Combine(1, 2),
+            HashCode.Combine(1, 2, 3),
+            HashCode.Combine(1, 2, 3, 4),
+            HashCode.Combine(1, 2, 3, 4, 5),
+            HashCode.Combine(1, 2, 3, 4, 5, 6),
+            HashCode.Combine(1, 2, 3, 4, 5, 6, 7),
+            HashCode.Combine(1, 2, 3, 4, 5, 6, 7, 8),
+
+            HashCode.Combine(2),
+            HashCode.Combine(2, 3),
+            HashCode.Combine(2, 3, 4),
+            HashCode.Combine(2, 3, 4, 5),
+            HashCode.Combine(2, 3, 4, 5, 6),
+            HashCode.Combine(2, 3, 4, 5, 6, 7),
+            HashCode.Combine(2, 3, 4, 5, 6, 7, 8),
+            HashCode.Combine(2, 3, 4, 5, 6, 7, 8, 9),
+        };
+
+        for (var i = 0; i < hcs.Length; i++)
+            for (var j = 0; j < hcs.Length; j++)
+            {
+                if (i == j) continue;
+                Assert.NotEqual(hcs[i], hcs[j]);
+            }
+    }
+
```

Comment:
In the issues' proposal, the idea was that building the hash-code using `.Add()` would result in the same hash-code as using a single `.Combine()` call; it would be nice to have a test-case for that.